File: go/fizzbuzz.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

type wordPair struct {
	word   string
	number int
}

// Words should be in count order
var words = [...]wordPair{{"Fizz", 3}, {"Buzz", 5}}

func main() {
	if len(os.Args) == 2 { // INT
		checkDivisible(stringToiOrError(os.Args[1]))
	} else if len(os.Args) == 3 { // MIN/MAX
		min, max := stringToiOrError(os.Args[1]), stringToiOrError(os.Args[2])
		for i := min; i <= max; i++ {
			checkDivisible(i)
		}
		if min >= max {
			abort("Error:Bad range "+os.Args[1]+" >= "+os.Args[2], 5)
		}
	} else {
		abort("Error: Incorrect number of Arguments", 3)
	}
}

// Exits program by sending error message to standard error and specified error code.
func abort(errorMessage string, exitcode int) {
	fmt.Fprintln(os.Stderr, errorMessage)
	os.Exit(exitcode)
}

// Attempts to convert sting to integer or aborts program
func stringToiOrError(arg string) int {
	i, err := strconv.Atoi(arg)
	if err != nil {
		abort("Error: "+arg+"is not an integer", 4)
	}
	return i
}

// Check if value is divisable to number and outputs word or number to standard out.
func checkDivisible(value int) {
	processed := ""
	for _, check := range words {
		if value%check.number == 0 {
			processed += check.word
		}
	}
	if processed == "" {
		fmt.Println(value)
	} else {
		fmt.Println(processed)
	}

}

File: cpp/fizzbuzz.cpp
#include <iostream>
#include <map>
#include <string>
#include <stdexcept> 

// Word List
const std::map<int, std::string> WORDS = {{3, "Fizz"}, {5, "Buzz"}};

// Attempt to convert a string into an int.
// It will exit program if the conversion fails.
int intCheckConvert(const std::string &value){
    int new_int{};
    try{
        new_int = std::stoi(value);
    } catch(const std::invalid_argument& _){
        std::cerr << "Error: `" << value << "` is not an integer" << std::endl;
        exit(4);
      } catch(const std::out_of_range& _) {
        std::cerr << "Error: `" << value << "` is out of range for an integer" << std::endl;
        exit(4);
      }
    return new_int;
}

// Check if value is divisable to number and outputs word or number to standard out.
void checkDivisible(int value){
  std::string processed {};
  for (auto& [key, word]: WORDS) if(value % key == 0) processed += word;
  if (processed.empty()) std::cout << value << std::endl;
  else std::cout << processed << std::endl;
}

int main(int argc, char* argv[]){
  // First argument is programs name.
  switch(argc){
    case 2: // One Number provided
      checkDivisible(intCheckConvert(argv[1]));
      break;
    case 3: { // two Number provided
      const int min = intCheckConvert(argv[1]);
      const int max = intCheckConvert(argv[2]);
      if (min >= max) {
        std::cerr << "Error: Bad range " << min << " >= " << max << std::endl;
        std::exit(5);
      }
      for(int i = min; i <= max; i++) checkDivisible(i);
    }
      break;
    default:
      std::cerr << "Error: Incorrect number of Arguments" << std::endl;
      std::exit(3);
      break;
  }
  return 0;
}